### PR TITLE
Auto update dependencies

### PR DESCRIPTION
**Pull request by RincewindsHat**
This PR adds:

* Automatic tests
* Automatic dependency updates
* Let git ignore compiled executable

### CODE REVIEW

File: firmware.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func stringFromReader(reader *bufio.Reader, file string, desc string) *string {
 		_log(nil, "Failed to read %v from %v: %v\n", desc, file, err)
 		return nil
 	}
-	str = strings.Trim(str, string(0))
+	str = strings.Trim(str, "0")
```
Comment (pzii) [inline firmware.go:115]: This modification changes the code's behavior and breaks parsing firmware files ("0" != "\x00"). I, however, agree that string(0) is not an elegant solution and should be replaced ;)

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Go
+on:
+  push:
+    tags:
+      - v*
+    branches: [ master ]
```
Comment (pzii) [inline .github/workflows/go.yml:6]: There is no _master_ branch - only _main_

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
```
Comment (pzii) [inline .github/workflows/golangci-lint.yml:5]: There is no _master_ branch - only _main_

File: .github/workflows/go.yml (hunk 2 above)
Comment (RincewindsHat) [reply .github/workflows/go.yml]: indeed, copy and paste error, fixed

File: .github/workflows/golangci-lint.yml (hunk 3 above)
Comment (RincewindsHat) [reply .github/workflows/golangci-lint.yml]: indeed, copy and paste error, fixed

File: firmware.go (hunk 1 above)
Comment (RincewindsHat) [reply firmware.go]: Removed my changes there

**Review by pzii — APPROVED** [review]
Nice - thank you :)

**Pull request merged**